src/parameterizations/lateral/MOM_tidal_forcing.F90

```diff
@@ -58,6 +67,159 @@ module MOM_tidal_forcing
 
 contains
 
+!> Finds astronomical longitudes s, h, p, and N,
+!! the mean longitude of the moon, sun, lunar perigee, and ascending node, respectively,
+!! at the specified reference time time_ref.
+!! These formulas were obtained from
+!! Kowalik and Luick, "Modern Theory and Practice of Tide Analysis and Tidal Power", 2019
+!! (their Equation I.71), which are based on Schureman, 1958.
+!! For simplicity, the time associated with time_ref should
+!! be at midnight. These formulas also only make sense if
+!! the calendar is gregorian.
+subroutine astro_longitudes_init(time_ref, longitudes_shpn)
+  real, dimension(4), intent(out) :: longitudes_shpn !> Longitudes s, h, p, N
+  real, intent(in) :: time_ref                       !> Time to calculate longitudes for.
+  real :: D, T                                       !> Date offsets
+  real, parameter :: PI = 4.0 * atan(1.0)            !> 3.14159...
+  ! if time_ref is not at midnight, this could be used to round down to nearest day.
+  ! time_ref = time_ref - mod(time_ref, 24.0*3600.0)
+  ! Find date at time_ref in days since 1900-01-01
+  D = (time_ref - time_type_to_real(set_date(1900, 1, 1))) / (24.0 * 3600.0)
```

Comment:
This would be more accurate if this were done as a time-type calculation, which would include support for a variety of different calendars.